lib/src/polygon_clipping/segment.dart

```diff
@@ -0,0 +1,573 @@
+// Give segments unique ID's to get consistent sorting of
+// segments and sweep events when all else is identical
+import 'dart:math';
+
```

Comment:
generally: strong types for parameters and return types of functions. Will take a look into this file in a future PR